core/dbt/events/functions.py

```diff
@@ -181,17 +153,8 @@ def create_json_log_line(e: T_Event, msg_fn: Callable[[T_Event], str]) -> str:
     else:
         values['data'] = None
 
-    # need to catch if any data is not serializable but still make sure as much of
-    # the logs go out as possible
-    try:
-        log_line = json.dumps(scrub_dict_secrets(values), sort_keys=True)
-    except TypeError:
-        # the only key currently throwing errors is 'data'.  Expand this list
-        # as needed if new issues pop up
-        values['data'] = None
-        values['event_data_serialized'] = False
-        log_line = json.dumps(scrub_dict_secrets(values), sort_keys=True)
-    return log_line
+    raw_log_line = json.dumps(values, sort_keys=True)
+    return scrub_secrets(raw_log_line, env_secrets())
```

Comment:
I think you may still need the try statement to build the raw log line.  Then scrub on the return.  The `data` field has unpredictable contents.

Reply:
The data field has very predictable contents! We wrote them all down in `types.py`. I think we just have to handle them better. I'll propose something here and see what you think.